pulsar-client/src/main/java/org/apache/pulsar/client/impl/MessageIdImpl.java

```diff
@@ -210,7 +210,8 @@ public int compareTo(MessageId o) {
         } else if (o instanceof TopicMessageIdImpl) {
             return compareTo(((TopicMessageIdImpl) o).getInnerMessageId());
         } else {
-            throw new UnsupportedOperationException("Unknown MessageId type: " + o.getClass().getName());
+            final String typeName = (o != null) ? o.getClass().getName() : "null";
```

Comment:
we can add a check in the beginning of the method.

```
if (o == null) {
   throw new UnsupportedOperationException(....)
}
```

Reply:
I agree.